impl-tinkerpop/src/test/java/org/hawkular/inventory/impl/tinkerpop/test/BasicTest.java

```diff
@@ -283,8 +295,12 @@ public void testRelationshipServiceGet() throws Exception {
             "going from  environments must have uid equal to 'test'. Was: " + contains.getTarget().getId();
         assert "contains" .equals(contains.getName()) : "Name of the relation must be 'contains'.";
 
-        contains = inventory.tenants().getAll().environments().get("test").relationships().get("13").entity();
-        assert contains == null : "There should not be any edge with id 13.";
+        try {
+            inventory.tenants().getAll().environments().get("test").relationships().get("13").entity();
```

Comment:
btw. I found much better to test some general properties of the graph than some implementation details. What about using something for checking those properties on randomly generated graphs? ScalaCheck test framework has so called generators, I think it's inspired by QuickCheck, that are able to generate random data structures (lists/trees/heaps/graphs..). 

And tests verifies some general properties like:
 * when adding 1 edge to a random graph, resulting graph has number of edges equal to the original graph + 1,

* when removing a tenant, all it's data are deleted (no leaking, we don't support it yet) and similar.

Here is something http://www.scala-graph.org/guides/test.html but it can be easily written for our needs.